paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/refund/PaytrailRefundPaymentService.java

```diff
@@ -0,0 +1,343 @@
+                    HttpStatus.FORBIDDEN,
+                    new Error("rejected-creating-payment-for-order-without-user", "rejected creating refund payment for refund without user, refundId id [" + refundId + "]")
+            );
+        }
+    }
+
+    public void isValidRefundStatusToCreateRefundPayment(String refundId, String refundStatus) {
+        // check refund status, can only create refund payment for confirmed refunds
+        if (!"confirmed".equals(refundStatus)) {
+            log.warn("creating refund payment for unconfirmed refund rejected, refundId: " + refundId);
+            throw new CommonApiException(
+                    HttpStatus.FORBIDDEN,
+                    new Error("rejected-creating-refund-payment-for-unconfirmed-refund", "rejected creating refund payment for unconfirmed refund, refundId id [" + refundId + "]")
+            );
+        }
+    }
+
+    public void setPaymentStatus(String paymentId, String status) {
+        RefundPayment payment = getRefund(paymentId);
+        payment.setStatus(status);
+        refundPaymentRepository.save(payment);
+    }
+
+    public RefundPayment getRefundPaymentForOrder(String orderId) {
+        List<RefundPayment> payments = refundPaymentRepository.findByOrderId(orderId);
+
+        RefundPayment paidPayment = selectPaidRefundPayment(payments);
+        RefundPayment payablePayment = selectPayableRefundPayment(payments);
+
+        if (paidPayment != null) {
+            return paidPayment;
+        } else if (payablePayment != null) {
+            return payablePayment;
+        } else {
+            return null;
+        }
+    }
+
+    public List<RefundPayment> getPaymentsForOrder(String orderId, String namepace) {
+        return refundPaymentRepository.findByNamespaceAndOrderId(namepace, orderId);
+    }
+
+    public List<RefundPayment> getPaymentsWithNamespaceAndOrderIdAndStatus(String orderId, String namepace, String status) {
+        return refundPaymentRepository.findByNamespaceAndOrderIdAndStatus(namepace, orderId, status);
+    }
+
+
+    private RefundPayment selectPayableRefundPayment(List<RefundPayment> payments) {
+        RefundPayment payablePayment = null;
+
+        if (payments != null)
+            for (RefundPayment payment : payments) {
+                // in an unpayable state
+                if (Objects.equals(payment.getStatus(), RefundPaymentStatus.PAID_ONLINE) || Objects.equals(payment.getStatus(), RefundPaymentStatus.CANCELLED)) {
+                    continue;
+                }
+
+                // an earlier selected refund payment is newer
+                if (payablePayment != null && payablePayment.getTimestamp().compareTo(payment.getTimestamp()) > 0) {
+                    continue;
+                }
+
+                payablePayment = payment;
+            }
+
+        return payablePayment;
+    }
+
+    private RefundPayment selectPaidRefundPayment(List<RefundPayment> payments) {
+        if (payments != null) {
```

Comment:
This is if -statement is actually not necessary, because refundPaymentRepository.findByOrderId(orderId) will never return null - spring data repository only returns empty array if no matches are found.

Reply:
I will remove this, I created automation test to test that, and no functionality changed.